.buildkite/metricbeat/pipeline.yml

```diff
@@ -10,43 +12,143 @@ env:
 
-
 steps:
+  - group: "Metricbeat Mandatory Tests"
+    key: "metricbeat-mandatory-tests"
+    steps:
+      - label: ":linux: Metricbeat Ubuntu Unit Tests"
+        key: "metricbeat-mandatory-linux-unit-test"
+        command: ".buildkite/scripts/unit_tests.sh"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_UBUNTU_X86_64}"
+          machineType: "${GCP_DEFAULT_MACHINE_TYPE}"
+        artifact_paths: "${BEATS_PROJECT_NAME}/build/*.*"
+
+      - label: ":go: Metricbeat Go Integration Tests"
+        key: "metricbeat-mandatory-int-test"
+        command: ".buildkite/scripts/go_int_tests.sh"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_UBUNTU_X86_64}"
+          machineType: "${GCP_HI_PERF_MACHINE_TYPE}"
+        artifact_paths: "${BEATS_PROJECT_NAME}/build/*.*"
+
+      - label: ":python: Metricbeat Python Integration Tests"
+        key: "metricbeat-mandatory-python-int-test"
+        command: ".buildkite/scripts/py_int_tests.sh"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_UBUNTU_X86_64}"
+          machineType: "${GCP_HI_PERF_MACHINE_TYPE}"
+        artifact_paths: "${BEATS_PROJECT_NAME}/build/*.*"
 
-  - input: "Input Parameters"
-    key: "run_metricbeat"
-    fields:
-    - select: "Metricbeat - run_metricbeat"
-      key: "run_metricbeat"
-      options:
-        - label: "True"
-          value: "true"
-        - label: "False"
-          value: "false"
-      default: "false"
-    - select: "Metricbeat - run_metricbeat_macos_tests"
-      key: "run_metricbeat_macos_tests"
-      options:
-        - label: "True"
-          value: "true"
-        - label: "False"
-          value: "false"
-      default: "false"
-    if: "build.source == 'ui'"
+      - label: ":negative_squared_cross_mark: Metricbeat Cross compile"
+        key: "metricbeat-mandatory-cross-compile"
+        command: ".buildkite/scripts/crosscompile.sh"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_UBUNTU_X86_64}"
+          machineType: "${GCP_DEFAULT_MACHINE_TYPE}"
+        artifact_paths: "${BEATS_PROJECT_NAME}/build/*.*"
+
+      - label: ":windows: Metricbeat Windows 2016/2022 Unit Tests - {{matrix.image}}"
+        command: ".buildkite/scripts/win_unit_tests.ps1"
+        key: "metricbeat-mandatory-win-unit-tests"
+        agents:
+          provider: "gcp"
+          image: "{{matrix.image}}"
+          machine_type: "${GCP_WIN_MACHINE_TYPE}"
+          disk_size: 100
+          disk_type: "pd-ssd"
+        matrix:
+          setup:
+            image:
+              - "${IMAGE_WIN_2016}"
+              - "${IMAGE_WIN_2022}"
+        artifact_paths: "${BEATS_PROJECT_NAME}/build/*.*"
+
+  - group: "Metricbeat Extended Windows Tests"
+    key: "metricbeat-extended-win-tests"
+    if: build.env("BUILDKITE_PULL_REQUEST") == "false" || build.env("GITHUB_PR_LABELS") =~ /.*windows.*/
+    steps:
+      - label: ":windows: Metricbeat Windows 2019 Unit Tests"
```

Comment:
wondering if this can be a matrix step, similar as above. 
I see the wind 10/2019/11 be the same.
Do we want to keep explicit steps or use matrixes?

Reply:
Let's keep it for now. I don't know what is better